compiler/fir/fir2ir/src/org/jetbrains/kotlin/fir/backend/Fir2IrDeclarationStorage.kt

```diff
@@ -262,6 +263,9 @@ class Fir2IrDeclarationStorage(
                 irDeclaration.parent = irClass
             }
         }
+        if (regularClass.origin == FirDeclarationOrigin.Library) {
```

Comment:
Seems the pattern is repeated several times (except for the propertyAccessor case). Maybe adding a util function?

Reply:
Yes, I noticed that I did more repetition while reviewing your PR. :D Updated, thanks!